keylime/src/list_parser.rs

```diff
@@ -0,0 +1,297 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Keylime Authors
+
+use core::{iter::Peekable, str::Chars};
+use log::*;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("ListParser: End of string reached unexpectedly")]
+    EOF,
+    #[error("ListParser: Missing closing ']' at position {0}")]
+    ClosingBracket(usize),
+    #[error("ListParser: Missing closing '{0}' at position {1}")]
+    ClosingQuote(char, usize),
+    #[error("ListParser: Empty string at position {0}")]
+    EmptyString(usize),
+}
+
+struct ListParser<'a> {
+    c: Peekable<Chars<'a>>,
+    consumed: usize,
+}
+
+impl<'a> ListParser<'a> {
+    fn parse(&mut self, bracketed: bool) -> Result<Vec<String>, Error> {
+        let mut r = Vec::new();
+
+        while let Some(c) = self.c.peek() {
+            match *c {
+                ',' | ' ' | '\n' | '\t' | '\r' => {
```

Comment:
Just wondering: how bad it would be to use a parsing library like [pest|https://pest.rs/], as some of them are already appear in the dependencies?

Reply:
This is a possibility I considered, but it felt like an overkill. If preferred, I can do it (the addition of the dependency as you pointed out is not very impactful as we have a transitive dependency).

Reply:
In the end, I like the result obtained by generating the parser with `pest`. It is easier to understand the intended grammar. Maybe there is a simpler grammar than what I used, but it covers the test cases I created for the manually written parser. The part in which the parsed elements are obtained from the resulting structure did not require sophisticated  processing as we simply want to extract the strings, not understand the structure of the provided list.